src/triage/component/architect/builders.py

```diff
@@ -24,10 +25,17 @@ def __init__(
     ):
         self.db_config = db_config
         self.matrix_directory = matrix_directory
-        self.engine = engine
+        self.serializable_db_engine = SerializableDbEngine(engine)
         self.replace = replace
         self.include_missing_labels_in_train_as = include_missing_labels_in_train_as
-        self.sessionmaker = sessionmaker(bind=self.engine)
+
+    @property
+    def sessionmaker(self):
+        return sessionmaker(bind=self.db_engine)
+
+    @property
+    def db_engine(self):
+        return self.serializable_db_engine.db_engine
```

Comment:
I assume the sessionmaker change is needed because a sessionmaker object isn't pickleable.

Is it by any chance pickleable when instantiated with the "serializable" engine version, however?

(Alternatively, I imagine it could be cached without pickling, but I doubt this is such a big deal.)

(See above/below regarding `engine` / `db_engine`.)